tidy3d/components/data/dataset.py

```diff
@@ -942,8 +997,17 @@ def interp(
             y-coordinates of sampling points.
         z : Union[float, ArrayLike]
             z-coordinates of sampling points.
-        fill_value : float = 0
-            Value to use when filling points without interpolated values.
+        fill_value : Union[float, Literal["extrapolate"]] = "extrapolate"
+            Value to use when filling points without interpolated values. If ``extrapolate`` then
+            nearest values are used.
+        use_vtk : bool = False
```

Comment:
Can you clarify here what are the differences between vtk's interpolation and our own interpolation? In terms of result or performance

Reply:
Also, `use_vtk` is only used if `method="linear"` I guess. My understanding is that `use_vtk` would only be used if the default behavior had performance issues? Maybe it makes sense to get rid of `use_vtk` and just have this interp function use our own interpolation (either interp_nearest or interp_py). Then, we could make interp_vtk a public function for the case when one does need to use vtk interpolation.